services-camel/src/test/java/edu/unc/lib/cdr/thumbnail/ThumbnailRouterTest.java

```diff
@@ -0,0 +1,227 @@
+import static edu.unc.lib.dl.rdf.Fcrepo4Repository.Binary;
+import static edu.unc.lib.dl.rdf.Premis.hasMessageDigest;
+import static org.apache.jena.rdf.model.ResourceFactory.createResource;
+import static org.fcrepo.camel.FcrepoHeaders.FCREPO_AGENT;
+import static org.fcrepo.camel.FcrepoHeaders.FCREPO_BASE_URL;
+import static org.fcrepo.camel.FcrepoHeaders.FCREPO_DATE_TIME;
+import static org.fcrepo.camel.FcrepoHeaders.FCREPO_EVENT_TYPE;
+import static org.fcrepo.camel.FcrepoHeaders.FCREPO_URI;
+import static org.mockito.Mockito.when;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import org.apache.camel.BeanInject;
+import org.apache.camel.EndpointInject;
+import org.apache.camel.Produce;
+import org.apache.camel.ProducerTemplate;
+import org.apache.camel.PropertyInject;
+import org.apache.camel.builder.AdviceWithRouteBuilder;
+import org.apache.camel.component.mock.MockEndpoint;
+import org.apache.camel.model.ModelCamelContext;
+import org.apache.camel.test.spring.CamelSpringTestSupport;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.ModelFactory;
+import org.apache.jena.rdf.model.Resource;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.context.support.AbstractApplicationContext;
+import org.springframework.context.support.ClassPathXmlApplicationContext;
+
+import edu.unc.lib.dl.fcrepo4.PIDs;
+import edu.unc.lib.dl.fcrepo4.Repository;
+import edu.unc.lib.dl.fedora.PID;
+
+public class ThumbnailRouterTest extends CamelSpringTestSupport {
+	private static final String EVENT_NS = "http://fedora.info/definitions/v4/event#";
+	private static final String EVENT_TYPE = "org.fcrepo.jms.eventType";
+	private static final String IDENTIFIER = "org.fcrepo.jms.identifier";
+	private static final String RESOURCE_TYPE = "org.fcrepo.jms.resourceType";
+	private static final long timestamp = 1428360320168L;
+	private static final String userID = "bypassAdmin";
+	private static final String userAgent = "curl/7.37.1";
+	private static final String fileID = "/file1";
+	private final String eventTypes = EVENT_NS + "ResourceCreation";
+	
+	@PropertyInject(value = "fcrepo.baseUri")
+	private static String baseUri;
+	
+	@EndpointInject(uri = "mock:fcrepo")
+	protected MockEndpoint resultEndpoint;
+	
+	@BeanInject(value = "repository")
+	private Repository repo;
+	
+	@Produce(uri = "direct:start")
+	protected ProducerTemplate template;
+	
+	@Before
+	public void init() {
+		PIDs.setRepository(repo);
+		when(repo.getFedoraBase()).thenReturn(baseUri);
+	}
+	
+	@Override
+	protected AbstractApplicationContext createApplicationContext() {
+		return new ClassPathXmlApplicationContext("/service-context.xml");
+	}
+	
+	@Test
+	public void testRouteStartSuccess() throws Exception {
+		getMockEndpoint("mock:fcrepo:{{fcrepo.baseUri}}").expectedMessageCount(1);
```

Comment:
Since the actual "success" outcome of this part of the route would be that the "direct:images" endpoint was reached, perhaps we should test for that in all of these tests?